### PR TITLE
Added the code for checking the inaccurate date and time in the drive…

### PR DESCRIPTION
**Pull request by guptamanav422**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Showing the popup to update date and time in driver side if date and time for driver is not correct 
[Proof Of Testing](https://drive.google.com/file/d/1G5Ay3IbtIoJ5qaJgklNhAhwsYWPyYLzz/view?usp=sharing)
[Figma Link](https://www.figma.com/proto/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=61783-15659&t=evdAA1vnrnftVMWI-0&scaling=scale-down&page-id=12738%3A667&starting-point-node-id=33906%3A2892&show-proto-sidebar=1)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Flow.purs [hunk 1]
```diff
@@ -110,9 +111,23 @@ checkVersion :: Int -> FlowBT String Unit
 checkVersion versioncode = do
   when (versioncode < (getLatestAndroidVersion (getMerchant unit))) $ do
     lift $ lift $ doAff do liftEffect hideSplash
+    modifyScreenState $ AppUpdatePopUpScreenType (\appUpdatePopUpScreenState → appUpdatePopUpScreenState {appUpdateOrInaccourateTimeAndDate = "version"})
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Flow.purs:114]: Can you make It ENUM For the types

File: Frontend/ui-driver/src/Flow.purs [hunk 2]
```diff
@@ -110,9 +111,23 @@ checkVersion :: Int -> FlowBT String Unit
...
+  (CurrentDateAndTimeRes current)  <- Remote.currentDateAndTimeBT ""
+  let currentTimeStamp = current.timestamp
+  let deviceDateTimeStamp = getCurrentTimeStamp unit
+  let timeDiff = (((fromMaybe ( toNumber 0) currentTimeStamp )) - deviceDateTimeStamp)
+  let timeDiffInMins = (timeDiff) / toNumber (1000 * 60)
+  let absTimeDiff = if timeDiffInMins < toNumber 0 then timeDiffInMins * toNumber (-1) else timeDiffInMins
+  when (absTimeDiff >= toNumber 60 ) do
+    modifyScreenState $ AppUpdatePopUpScreenType (\appUpdatePopUpScreenState -> appUpdatePopUpScreenState { appUpdateOrInaccourateTimeAndDate = "dateAndTime" })
+    lift $ lift $ doAff do liftEffect hideSplash
+    void $ UI.handleAppUpdatePopUp
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Flow.purs:129]: can you rename this function `handleAppUpdatePopUp`

File: Frontend/ui-driver/src/Screens/AppUpdatePopUpScreen/Controller.purs [hunk 3]
```diff
@@ -34,17 +39,22 @@ instance loggableAction :: Loggable Action where
...
 
 eval :: Action -> AppUpdatePopUpScreenState -> Eval Action ScreenOutput AppUpdatePopUpScreenState
 eval OnCloseClick state = do
     exit Decline 
 eval OnAccept state = do 
     exit Accept 
-
+eval (PrimaryButtonActionController (PrimaryButton.OnClick)) state = do 
+  _ <- pure $ JB.launchDateSettings ""
+  _ <- pure $ setValueToLocalStore LAUNCH_DATE_SETTING "true"
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/AppUpdatePopUpScreen/Controller.purs:56]: Can you move this line above

File: Frontend/ui-driver/src/Screens/AppUpdatePopUpScreen/View.purs [hunk 4]
```diff
@@ -32,20 +33,96 @@ import Font.Size as FontSize
...
-view :: forall w .  (Action  -> Effect Unit) -> ST.AppUpdatePopUpScreenState -> PrestoDOM (Effect Unit) w
+
+view :: forall w . (Action  -> Effect Unit) -> ST.AppUpdatePopUpScreenState -> PrestoDOM (Effect Unit) w
 view push state =
-   linearLayout [
+  linearLayout
+  [ height MATCH_PARENT
+    , width MATCH_PARENT
+  ]
+  [ if (state.appUpdateOrInaccourateTimeAndDate == "version") then  updateRequiredView push state
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/AppUpdatePopUpScreen/View.purs:56]: Try to make it ENUM type

File: Frontend/ui-driver/src/Services/Endpoint.purs [hunk 5]
```diff
@@ -150,3 +150,6 @@ onCall _ = (getBaseUrl "") <> "/onCall"
 
 likeMessage :: String -> String
 likeMessage messageId = (getBaseUrl "") <> "/message/" <> messageId <> "/like"
+
+currentDateAndTime :: String -> String
+currentDateAndTime _ = "https://tools.aimylogic.com/api/now"
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Services/Endpoint.purs:155]: @witcher-shailesh  Can you verify this API Can be used?

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Engineering/Helpers/Commons.js [hunk 6]
```diff
@@ -342,6 +342,11 @@ export const getExpiryTime = function (str1) {
   };
 };
 
+export const getCurrentTimeStamp = function () {
+  console.log("current date and time "+(new Date()));
```
Comment (MercyQueen) [inline Frontend/ui-common/src/Engineering/Helpers/Commons.js:346]: @ManavGuptaJuspay can you please remove logs

File: Frontend/ui-driver/src/App.purs [hunk 7]
```diff
@@ -121,7 +121,7 @@ defaultGlobalState = GlobalState{
 , noInternetScreen : {}
 , popUpScreen : PopUpScreenData.initData
 , driverRideRatingScreen : DriverRideRatingScreenData.initData
-, appUpdatePopUpScreen : {version : 1}
+, appUpdatePopUpScreen : {version : 1, appUpdateOrInaccourateTimeAndDate : ""}
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/App.purs:124]: @ManavGuptaJuspay can you please correct the spelling of this variable or change the name

File: Frontend/ui-driver/src/Flow.purs [hunk 8]
```diff
@@ -72,12 +72,15 @@ import Merchant.Utils(getMerchant, Merchant(..))
 import Screens.RideSelectionScreen.View (getCategoryName)
 import Types.App (REPORT_ISSUE_CHAT_SCREEN_OUTPUT(..), RIDES_SELECTION_SCREEN_OUTPUT(..), ABOUT_US_SCREEN_OUTPUT(..), BANK_DETAILS_SCREENOUTPUT(..), ADD_VEHICLE_DETAILS_SCREENOUTPUT(..), APPLICATION_STATUS_SCREENOUTPUT(..), DRIVER_DETAILS_SCREEN_OUTPUT(..), DRIVER_PROFILE_SCREEN_OUTPUT(..), DRIVER_RIDE_RATING_SCREEN_OUTPUT(..), ENTER_MOBILE_NUMBER_SCREEN_OUTPUT(..), ENTER_OTP_SCREEN_OUTPUT(..), FlowBT, GlobalState(..), HELP_AND_SUPPORT_SCREEN_OUTPUT(..), HOME_SCREENOUTPUT(..), MY_RIDES_SCREEN_OUTPUT(..), NOTIFICATIONS_SCREEN_OUTPUT(..), NO_INTERNET_SCREEN_OUTPUT(..), PERMISSIONS_SCREEN_OUTPUT(..), POPUP_SCREEN_OUTPUT(..), REGISTRATION_SCREENOUTPUT(..), RIDE_DETAIL_SCREENOUTPUT(..), SELECT_LANGUAGE_SCREEN_OUTPUT(..), ScreenStage(..), ScreenType(..), TRIP_DETAILS_SCREEN_OUTPUT(..), UPLOAD_ADHAAR_CARD_SCREENOUTPUT(..), UPLOAD_DRIVER_LICENSE_SCREENOUTPUT(..), VEHICLE_DETAILS_SCREEN_OUTPUT(..), WRITE_TO_US_SCREEN_OUTPUT(..), NOTIFICATIONS_SCREEN_OUTPUT(..), REFERRAL_SCREEN_OUTPUT(..), BOOKING_OPTIONS_SCREEN_OUTPUT(..), defaultGlobalState)
 import Screens.HomeScreen.ComponentConfig (specialLocationConfig)
+import Debug
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Flow.purs:75]: @ManavGuptaJuspay please remove this import and also spy statements if they are being used

File: Frontend/android-native/app/src/driver/assets/juspay/juspay_assets.json [hunk 9]
```diff
@@ -36,22 +38,11 @@
...
-      "ic_call_plus" : true,
-      "ic_cancelride_metro_drop" : true,
-      "ic_cancelride_metro_pickup" : true,
-      "ic_chat" : true,
-      "ic_chat_badge" : true,
-      "ic_chat_blue" : true,
-      "ic_driver_status_offline" : true,
-      "ic_driver_status_online" : true,
-      "ic_driver_status_silent" : true,
       "ic_graph_black" : true,
```
Comment (MercyQueen) [inline Frontend/android-native/app/src/driver/assets/juspay/juspay_assets.json:44]: @ManavGuptaJuspay why are these images removed from juspay_assets

**Review by MercyQueen — CHANGES_REQUESTED** [review]

**Review by aryan-13 — APPROVED** [review]

**Review by MercyQueen — APPROVED** [review]

File: Frontend/ui-driver/src/Services/Endpoint.purs (hunk 5 above)
Comment (witcher-shailesh) [reply Frontend/ui-driver/src/Services/Endpoint.purs:161]: What is API latency here?

**Pull request merged**